src/Workspaces/Core/Portable/Differencing/LongestCommonSubsequence.cs

```diff
@@ -179,6 +179,15 @@ protected double ComputeDistance(TSequence oldSequence, int oldLength, TSequence
                 return (oldLength == newLength) ? 0.0 : 1.0;
             }
 
+            // If the sequences differ significantly in size their distance will be very close to 1.0 
+            // (even if one is a strict subsequence of the other).
+            // Avoid running an expensive LCS algorithm on such sequences.
+            double lenghtRatio = (oldLength > newLength) ? oldLength / newLength : newLength / oldLength;
+            if (lenghtRatio > 100)
```

Comment:
>100 [](start = 30, length = 3)

100 is good. Is it safe and reasonable to make this number smaller?

Reply:
1.0 - 1/100 = 0.99 

If we make it 10 the result without this optimization would be something around 0.90. I think we should keep the precision to 2 digits, since we multiply this distance by numbers in range 0.1-0.9 when calculating weighted distances.